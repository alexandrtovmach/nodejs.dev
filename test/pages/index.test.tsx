import React from 'react';
import Landing from '../../src/pages/index';
import * as ShallowRenderer from 'react-test-renderer/shallow';

describe('Landing page', () => {
  it('renders correctly', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<Landing />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
