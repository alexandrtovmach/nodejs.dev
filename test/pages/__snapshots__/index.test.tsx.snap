// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page renders correctly 1`] = `
<Layout
  description="Welcome to Node.js!"
  title="Node.js Official website"
  withBg={true}
>
  <MainSection
    content={
      Object {
        "desc": "Node.js is a free, open-sourced, cross-platform JavaScript run-time environment that lets developers write command line tools and server-side scripts outside of a browser.",
        "downloadButton": "Download Node.js (LTS)",
        "downloadCurrent": "Get Current",
        "learnNode": "Learn Node.js",
        "subDownloadMessage": "What’s new",
        "title": "Run JavaScript Everywhere",
      }
    }
  />
  <SandboxSection />
  <FeaturesSection
    content={
      Object {
        "featureList": Array [
          Object {
            "image": "test-file-stub",
            "text": "Feat #1 Cornhole lomo dreamcatcher celiac enamel pin, yr lyft master cleanse meggings.",
          },
          Object {
            "image": "test-file-stub",
            "text": "Feat #2 Tacos raw denim palo santo, squid 90 occupy sustainable leggings locavore before they sold out chambray gastropub synth.",
          },
          Object {
            "image": "test-file-stub",
            "text": "Feat #3 Lorem ipsum dolor amet pug vape +1 poke pour-over kitsch tacos meh. ",
          },
        ],
      }
    }
  />
  <CommunitySection
    content={
      Object {
        "desc": "We’ll never share your information and always respect your inbox - quality content only, we promise.",
        "eventsLink": "See events near you",
        "subscribe": "Subscribe",
        "title1": "Join",
        "title2": "the community",
      }
    }
  />
  <PartnerSection
    content={
      Object {
        "desc": "Including IBM, LinkedIn, Microsoft, Netflix, and PayPal.",
        "partnersLogoList": Array [
          "test-file-stub",
          "test-file-stub",
          "test-file-stub",
          "test-file-stub",
          "test-file-stub",
        ],
        "title": "Trusted by development teams around the world",
      }
    }
  />
  <GetStartedSection
    content={
      Object {
        "actionButton": "Get Started",
        "blocks": Array [
          Object {
            "desc": "Lorem ipsum dolor amet pug vape +1 poke pour-over kitsch tacos meh.",
            "image": "test-file-stub",
            "link": "/learn",
            "title": "A beginner’s guide",
          },
          Object {
            "desc": "Tacos raw denim palo santo, squid 90 occupy sustainable leggings locavore before they sold out chambray gastropub synth.",
            "image": "test-file-stub",
            "link": "/docs",
            "title": "Do even more with Node.js",
          },
        ],
      }
    }
  />
</Layout>
`;
